src/plugin/managerbase.js

```diff
@@ -0,0 +1,471 @@
+                    if (!pluginContext.commitHash) {
+                        throw new Error('Neither commitHash nor branchHash from branch was obtained, branchName: [' +
+                            context.branchName + ']');
+                    }
+                    return Q.ninvoke(pluginContext.project, 'loadObject', pluginContext.commitHash);
+                })
+                .then(function (commitObject) {
+                    var rootDeferred = Q.defer();
+                    self.logger.debug('commitObject loaded', {metadata: commitObject});
+                    pluginContext.core.loadRoot(commitObject.root, function (err, rootNode) {
+                        if (err) {
+                            rootDeferred.reject(err);
+                        } else {
+                            self.logger.debug('rootNode loaded');
+                            rootDeferred.resolve(rootNode);
+                        }
+                    });
+
+                    return rootDeferred.promise;
+                })
+                .then(function (rootNode) {
+                    pluginContext.rootNode = rootNode;
+                    // Load active node
+                    return self.loadNodeByPath(pluginContext, context.activeNode || '');
+                })
+                .then(function (activeNode) {
+                    pluginContext.activeNode = activeNode;
+                    self.logger.debug('activeNode loaded');
+                    // Load active selection
+                    return self.loadNodesByPath(pluginContext, context.activeSelection || []);
+                })
+                .then(function (activeSelection) {
+                    pluginContext.activeSelection = activeSelection;
+                    self.logger.debug('activeSelection loaded');
+                    // Load meta nodes
+                    var metaIds = pluginContext.core.getMemberPaths(pluginContext.rootNode, 'MetaAspectSet');
+                    return self.loadNodesByPath(pluginContext, metaIds, !context.namespace);
+                })
+                .then(function (metaNodes) {
+                    var libraryNames = pluginContext.core.getLibraryNames(pluginContext.rootNode),
+                        metaNodeName,
+                        nodeNamespace,
+                        i;
+
+                    function startsWith(str, pattern) {
+                        return str.indexOf(pattern) === 0;
+                    }
+
+                    if (context.namespace) {
+                        if (libraryNames.indexOf(context.namespace) === -1) {
+                            throw new Error('Given namespace does not exist among the available: "' +
+                                libraryNames + '".');
+                        }
+
+                        for (i = 0; i < metaNodes.length; i += 1) {
+                            nodeNamespace = pluginContext.core.getNamespace(metaNodes[i]);
```

Comment:
That could lead to collisions. If a meta node, in a project with a library named `A`, starts with `A.` - it would appear in the `A` namespace.